### ISSUE TITLE
File size of QA Tiles generated at level 14

### ISSUE THREAD

**Issue by jasonbeverage** [mapbox/tippecanoe#518]
Hi!

I've been generating QA tiles at level 14 (instead of the 12 that mapbox publishes for it's historical datasets) using minjur and tippecanoe for awhile and I recently tried replacing minjur with the new osmium export command to get the better multi-polygon support and drop the unmaintained minjur command.

Previously, I was using a minjur/tippecanoe command line like this:
minjur -n dense -p planet.pbf | tippecanoe -l osm -Z 14 -z 14 -b 0 -o osm-qa-14.mbtiles

The output mbtiles file was just slightly bigger than the input planet file.  So if the input planet.pbf was ~40GB then the output mbtiles was like 43GB.  Now I know that my command line isn't preventing features from being dropped and it's also not outputting multipolygons, so it's not the FULL osm dataset.

When I use osmium export and change my tippecanoe command line to match what is in the mapbox qa tiles [page](http://osmlab.github.io/osm-qa-tiles/historic.html), the output mbtiles file is nearly 500GB.

This is using this command line:
osmium export -f geojsonseq --config=config.json planet.osm.pbf | tippecanoe -f -pf -pk -ps -Z14 -z14 -d12 --no-tile-stats -b0 -l osm -n osm-qa -o osm-qa.mbtiles

The config.json I'm using is here:  https://github.com/osmcode/osmium-tool/blob/master/export-example-config/example-config.json

If I change to using -Z12 and -z12 instead of 14, the output mbtiles is more in line with what you provide on the QA tiles page, so around 45GB.

So my question is why is the output file size so big compared to what I was doing before with vanilla minjur at level 14?  And why is the file size SO much bigger at level 14 than 12 using the osmium-export command.

Possible reasons I came up with were:
1)  My original minjur command wasn't including multipolygons, but are there really that many multipolygons in the osm database?  My guess is that there might be :)
2)  My original tippecanoe command would drop features if there were too many in a tile.  I wouldn't expect that to happen very often at level 14, but maybe it is.
3)  There are exponentially more tiles in level 14 vs level 12, and they are smaller.  So those smaller tiles won't take advantage of pbf and zlib compression as well as the larger level 12 tiles.  Also since there are so many more tiles sqlite indices will be much larger as well.

I may have answered my own question, but I'd like a sanity check from the experts ;)

Also for reference the reason we're using level 14 instead of 12 is just to reduce the number of features in each tile.  We're using the output tileset in a simulator based on our [osgEarth](https://github.com/gwaldron/osgearth) globe rendering toolkit.

![osm-osgearth](https://user-images.githubusercontent.com/326626/35159525-8adeb984-fd08-11e7-8a37-931fc8d5d5ad.jpg)

**Comment by ericfischer** [comment]
I should test this myself to make sure, but I think the difference is that `minjur` wasn't producing Polygon geometries at all before, only LineStrings representing their boundaries. So I bet you are ending up now with huge numbers of identical tiles of water polygons, instead of just the tiles that touch the edge of the polygons.

**Comment by jasonbeverage** [comment]
Hey @ericfischer !

I'm fairly sure minjur did create polygons before with the -p flag b/c we were depending on that capability.  We actually ran into issues when we tried to use minjur-mp b/c it was outputting both polygons and linestrings for the same features.

That's a good guess about the water though.  I'll see if I can filter out water completely and do some tests on it.

**Comment by jasonbeverage** [comment]
Just to follow up on this, the issue appears to be the very geographically large polygons in the osm dataset that weren't being emitted previously with minjur.  The big ones I've identified are place=continent and place=state.  I've got a dataset now where I've just filtered out only what we actually use for visualization and it's much smaller.  You were definitely onto something with the issue though, you do end up with millions of tiny tiles for those giant polygons.

Thanks for your help in tracking down my issue!
